core/src/main/java/tc/oc/pgm/api/map/MapSource.java

```diff
@@ -39,10 +39,11 @@ public interface MapSource {
   /**
    * Download the {@link org.bukkit.World} files to a local directory.
    *
+   * @param folder subfolder to download, null for parent
    * @param dir An existent, but empty directory.
    * @throws MapMissingException If an error occurs while creating the files.
    */
-  void downloadTo(File dir) throws MapMissingException;
+  void downloadTo(String folder, File dir) throws MapMissingException;
```

Comment:
I don't think we should change the signature here, can the callee do the resolving?

Reply:
The MapSource is created *before* the map has been parsed, so the world folder to use isn't known at the time, and i'd rather keep MapSource immutable